File: repayment-server/src/main/java/com/example/repayment_server/dto/RepaymentResponseDto.java
package com.example.repayment_server.dto;

import com.example.repayment_server.constants.CommunicationStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@AllArgsConstructor
@Builder
@Getter
@Setter
public class RepaymentResponseDto {
    private Long repaymentId;
    private Long applicationId;
    private BigDecimal repaymentAmount;
    private BigDecimal balance;
    private CommunicationStatus communicationStatus;
    private LocalDateTime createdAt;
    private String createdBy;
    private LocalDateTime updatedAt;
    private String updatedBy;
}

File: ms-loan-frontend/src/app/dtos/repayment-response-dto.ts
import { CommunicationStatus } from './communitation-status';

export interface RepaymentResponseDto {
  repaymentId?: number;
  applicationId?: number;
  repaymentAmount?: number;
  balance?: number;
  communicationStatus?: CommunicationStatus;
  createdAt?: string;
  createdBy?: string;
}
